perun-core/src/main/java/cz/metacentrum/perun/core/bl/AttributesManagerBl.java

```diff
@@ -491,6 +491,18 @@
 	 */
 	List<Attribute> getAttributes(PerunSession sess, Host host) throws InternalErrorException;
 
+	/**
+	 * Get all attributes associated with the host which have name in list attrNames (empty and virtual too).
+	 *
```

Comment:
Add information to javadoc, that empty list will return no attributes. In some specific cases we have exactly different behavior (where empty means all). Same for other layers (impl, api, rpc).